src/test/java/by/itechart/api/service/impl/UserServiceImplTest.java

```diff
@@ -0,0 +1,158 @@
+        assertThat(resultDTO.getFirstName()).isEqualTo(userEntity.getFirstName());
+        assertThat(resultDTO.getLastName()).isEqualTo(userEntity.getLastName());
+        assertThat(resultDTO.getEmail()).isEqualTo(userEntity.getEmail());
+    }
+
+    @DisplayName("Update existing user")
+    @Test
+    void testUpdateExistingUser() {
+        Long id = 1L;
+        UpdateUserDTO userDTO = new UpdateUserDTO();
+        userDTO.setEmail("karen@karen.by");
+        userDTO.setFirstName("karen");
+        userDTO.setLastName("bagratyan");
+        User userEntity = new User();
+        when(userRepository.findById(id)).thenReturn(of(userEntity));
+        UserDTO resultUserDTO = userService.update(id, userDTO);
+        assertThat(resultUserDTO.getEmail()).isEqualTo(userDTO.getEmail());
+        assertThat(resultUserDTO.getFirstName()).isEqualTo(userDTO.getFirstName());
+        assertThat(resultUserDTO.getLastName()).isEqualTo(userDTO.getLastName());
+        verify(userRepository).saveAndFlush(any());
+    }
+
+    @Test
+    @DisplayName("Delete existing user")
+    void testDeleteExistingUser() {
+        Long id = 1L;
+        User userEntity = new User();
+        when(userRepository.findById(id)).thenReturn(of(userEntity));
+        userService.delete(id);
+        assertThat(userEntity.getDeletedAt()).isNotNull();
+    }
+
+    @Test
+    @DisplayName("Find all users")
+    void testFindAllUsers() {
+        List<User> suggesterAllUsers = new ArrayList<>();
+        User userEntityOne = new User();
+        User userEntityTwo = new User();
+        suggesterAllUsers.add(userEntityOne);
+        suggesterAllUsers.add(userEntityTwo);
+        when(userRepository.findAll()).thenReturn(suggesterAllUsers);
+        List<UserDTO> resultAllUsers = userService.findAll();
+        assertThat(resultAllUsers).isNotEmpty();
+    }
+
+    @Test
+    @DisplayName("Get current user")
+    void testGetCurrentUser() {
+        User userEntity = new User();
+        userEntity.setEmail("karen@karen.by");
+        userEntity.setFirstName("karen");
+        userEntity.setLastName("bagratyan");
+        when(userRepository.findByEmail(any())).thenReturn(of(userEntity));
+        UserDTO expectedDTO = userService.getCurrentUser(authentication);
+        assertThat(expectedDTO).isNotNull();
+    }
+
+    @Test
+    @DisplayName("Update user but user not found")
+    void testUpdateUser_UserNotFound_ExceptionThrown() {
+        Long notExistedIDForUser = 2L;
+        UpdateUserDTO userDTO = new UpdateUserDTO();
+        userDTO.setEmail("karen@karen.by");
+        userDTO.setFirstName("karen");
+        userDTO.setLastName("bagratyan");
+        Exception exception = assertThrows(UserNotFoundException.class,
+                () -> userService.update(notExistedIDForUser, userDTO));
+        String expectedMessage = "User with ID 2 is not found";
+        String actualMessage = exception.getMessage();
+        assertTrue(actualMessage.contains(expectedMessage));
```

Comment:
You can use[ AssertJ methods](https://www.baeldung.com/assertj-exception-assertion) to keep this concise